test/test_xnnpack_integration.py

```diff
@@ -711,71 +713,163 @@ class TestXNNPACKConv1dTransformPass(TestCase):
+        xnnpack_result = deserialized_optimized_scripted_model(input_data)
+        torch.testing.assert_allclose(ref_result, xnnpack_result, rtol=1e-2, atol=1e-3)
 
-        data_shape = (batch_size, input_channels, height)
-        pattern_count_map = {"Tensor = aten::conv1d": -1,
-                             "Tensor = aten::conv2d": 1}
-        TestXNNPACKConv1dTransformPass.validate_transform_conv1d_to_conv2d(Conv1D(), pattern_count_map, data_shape)
+    def test_conv1d_basic(self):
+        batch_size_list = range(1, 3)
+        input_channels_per_group_list = range(10, 12)
+        width_list = range(10, 12)
+        output_channels_per_group_list = range(10, 12)
+        groups_list = range(1, 3)
+        kernel_list = range(1, 4)
+        stride_list = range(1, 3)
+        padding_list = range(0, 3)
+        dilation_list = range(1, 3)
+
+        for hparams in itertools.product(batch_size_list,
+                                         input_channels_per_group_list,
+                                         width_list,
+                                         output_channels_per_group_list,
+                                         groups_list,
+                                         kernel_list,
+                                         stride_list,
+                                         padding_list,
+                                         dilation_list):
+            batch_size, input_channels_per_group, width, output_channels_per_group, \
+                groups, kernel, stride, padding, dilation = hparams
+
+            input_channels = input_channels_per_group * groups
+            output_channels = output_channels_per_group * groups
+            conv_weight_shape = (output_channels, input_channels_per_group, kernel)
+            conv_bias_shape = (output_channels)
+
+            class Conv1D(torch.nn.Module):
+                def __init__(self):
+                    super(Conv1D, self).__init__()
+                    self.weight = torch.nn.Parameter(torch.Tensor(torch.rand(conv_weight_shape)), requires_grad=False)
+                    self.bias = torch.nn.Parameter(torch.Tensor(torch.rand(conv_bias_shape)), requires_grad=False)
+                    self.stride = stride
+                    self.padding = padding
+                    self.dilation = dilation
+                    self.groups = groups
+
+                def forward(self, x):
+                    return F.conv1d(x, self.weight, self.bias,
+                                    self.stride, self.padding, self.dilation, self.groups)
+
+            data_shape = (batch_size, input_channels, width)
+            pattern_count_transformed_map = {"Tensor = aten::conv1d": -1,
+                                            "Tensor = aten::conv2d": 1}
+            pattern_count_optimized_map = {"Tensor = aten::conv1d": -1,
+                                            "Tensor = aten::conv2d": -1}
```

Comment:
Also add check for prepacked::conv2d_clamp_run or something like that. See other tests for the pattern.